M2/Macaulay2/m2/methods.m2

```diff
@@ -628,6 +628,7 @@ addHook(MutableHashTable, Thing, Function) := opts -> (store, key, hook) -> (
     store = store#key;
     ind := #store.HookPriority; -- index to add the hook in the list; TODO: use Priority to insert in the middle?
     alg := if opts.Strategy =!= null then opts.Strategy else ind;
+    if not store.HookAlgorithms#?alg then
```

Comment:
What if someone wants to overwrite a strategy?

Reply:
This indeed allows them to overwrite a strategy. This will prevent duplicating it, so the same strategy isn't attempted twice. To change the order they have to manually edit the priority list.

Reply:
Ah, ok.  I imagined a set of parentheses that wasn't there.  :)